src/Platform/Microsoft.Testing.Platform/Requests/TreeNodeFilter/TreeNodeFilter.cs

```diff
@@ -441,7 +441,13 @@ public bool MatchesFilter(string testNodeFullPath, PropertyBag filterablePropert
             if (currentFragmentIndex >= _filters.Count)
             {
                 // Note: The regex for ** is .*.*, so we match against such a value expression.
-                return currentFragmentIndex > 0 && _filters.Last() is ValueExpression { Value: ".*.*" };
+                FilterExpression lastFilter = _filters.Last();
+                if (lastFilter is ValueAndPropertyExpression valueAndPropertyExpression)
+                {
+                    lastFilter = valueAndPropertyExpression.Value;
+                }
+
+                return currentFragmentIndex > 0 && lastFilter is ValueExpression { Value: ".*.*" };
```

Comment:
I would need to make some tests and read the rest of the code but this is usually not recommended regex as it's a double eat all. It was there before so I won't block but it'd be good to investigate this part. Would you mind creating a ticket and assigning it to me?

Reply:
@Evangelink For context, I think the way the code is done like this is to differentiate `*` and `**`. `TokenizeFilter` will just substitute `*` with `.*`, then we are able to special case `.*.*` and are sure it was `**`.

I'll create an issue.

Reply:
#4496